src/prefect/cli/cloud.py

```diff
@@ -319,10 +319,17 @@ async def login(
 
     profiles = load_profiles()
     current_profile = get_settings_context().profile
-
-    if key and PREFECT_API_KEY.value() == key:
+    current_profile_api_key = current_profile.settings.get(PREFECT_API_KEY)
```

Comment:
This will bypass callbacks, you should use `settings.value_of(SETTING)` or `SETTING.value_from(settings)